ding/world_model/base_world_model.py

```diff
@@ -334,9 +331,9 @@ def rollout(self, obs: Tensor, actor_fn: Callable[[Tensor], Tuple[Tensor, Tensor
             torch.stack(obss),
             torch.stack(actions),
             # rewards is an empty list when horizon=0
-            torch.stack(rewards) if rewards else torch.tensor(rewards, device=device),
+            torch.stack(rewards) if rewards else torch.tensor(rewards, device=obs.device),
             torch.stack(aug_rewards),
-            torch.stack(dones)
+            torch.stack(dones) if dones else torch.tensor(dones, device=obs.device)
```

Comment:
maybe reward.device can be more general, because obs often owns diverse structures

Reply:
I can not change to `reward.device`, because MBRL starts only with `obs` and then do rollouts. And in MBRL, I do not think it is common that `obs` will own diverse structures because in `DreamWorldModel`, `obs` have to be an output from networks.